[Fix] Scheduler 모듈 GPT Client Bean 생성 실패 (openai.token 미설정)

## Bug Description

* 애플리케이션 실행 시 `gptClient` 빈 생성 과정에서 `gptConfig` 의존성 주입 실패 발생
* 원인: `application.yml` 또는 환경변수에서 `${openai.token}` 값을 읽지 못함

## Steps to Reproduce

1. `csalgo-scheduler` 모듈 실행
2. `gptClient` Bean 생성 시도
3. ApplicationContext 초기화 실패

## Expected Behavior

* `openai.token` 프로퍼티가 정상적으로 주입되어 `gptConfig`와 `gptClient` 빈이 정상 생성되어야 함

## Current Behavior

* `${openai.token}` placeholder를 해석하지 못해 `PlaceholderResolutionException` 발생
* 애플리케이션 컨텍스트 초기화가 중단됨

## Possible Solution

* `application.yml` 또는 `application-<profile>.yml`에 `openai.token` 속성 추가
* Kubernetes Secret 또는 환경변수 설정 시 `OPENAI_TOKEN` 값 주입 확인
* `@ConfigurationProperties` 또는 `@Value` 어노테이션 기반 설정 클래스(`GptConfig`)에서 매핑 여부 점검

## Additional Context

* 관련 에러 로그:

  ```
  Caused by: org.springframework.util.PlaceholderResolutionException: Could not resolve placeholder 'openai.token' in value "${openai.token}"
  ```

--- a/csalgo-iac/main.tf
+++ b/csalgo-iac/main.tf
@@ -30,7 +30,9 @@ module "server" {
 
   sentry_dsn = var.sentry_dsn
   jwt_secret = var.jwt_secret
+
   huggingface_api_token = var.huggingface_api_token
+  openai_api_token = var.openai_api_token
 }
 
 module "web" {
@@ -62,6 +64,9 @@ module "scheduler" {
 
   sentry_dsn = var.sentry_dsn
   jwt_secret = var.jwt_secret
+
+  huggingface_api_token = var.huggingface_api_token
+  openai_api_token = var.openai_api_token
 }
 
 module "dns" {

--- a/csalgo-iac/modules/scheduler/main.tf
+++ b/csalgo-iac/modules/scheduler/main.tf
@@ -16,6 +16,8 @@ resource "kubernetes_secret" "csalgo_scheduler_env" {
     REDIS_HOST    = var.redis_host
     REDIS_PORT    = var.redis_port
     JWT_SECRET    = var.jwt_secret
+    HUGGINGFACE_API_TOKEN = var.huggingface_api_token
+    OPENAI_API_TOKEN = var.openai_api_token
   }
 
   type = "Opaque"
@@ -54,123 +56,9 @@ resource "kubernetes_deployment" "csalgo_scheduler" {
           name  = "csalgo-scheduler"
           image = var.image
 
-          env {
-            name = "MAIL_HOST"
-            value_from {
-              secret_key_ref {
-                name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
-                key  = "MAIL_HOST"
-              }
-            }
-          }
-
-          env {
-            name = "MAIL_PORT"
-            value_from {
-              secret_key_ref {
-                name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
-                key  = "MAIL_PORT"
-              }
-            }
-          }
-
-          env {
-            name = "MAIL_USERNAME"
-            value_from {
-              secret_key_ref {
-                name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
-                key  = "MAIL_USERNAME"
-              }
-            }
-          }
-
-          env {
-            name = "MAIL_PASSWORD"
-            value_from {
-              secret_key_ref {
-                name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
-                key  = "MAIL_PASSWORD"
-              }
-            }
-          }
-
-          env {
-            name = "DB_HOST"
-            value_from {
-              secret_key_ref {
-                name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
-                key  = "DB_HOST"
-              }
-            }
-          }
-
-          env {
-            name = "DB_NAME"
-            value_from {
-              secret_key_ref {
-                name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
-                key  = "DB_NAME"
-              }
-            }
-          }
-
-          env {
-            name = "DB_USERNAME"
-            value_from {
-              secret_key_ref {
-                name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
-                key  = "DB_USERNAME"
-              }
-            }
-          }
-
-          env {
-            name = "DB_PASSWORD"
-            value_from {
-              secret_key_ref {
-                name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
-                key  = "DB_PASSWORD"
-              }
-            }
-          }
-
-          env {
-            name = "SENTRY_DSN"
-            value_from {
-              secret_key_ref {
-                name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
-                key  = "SENTRY_DSN"
-              }
-            }
-          }
-
-          env {
-            name = "REDIS_HOST"
-            value_from {
-              secret_key_ref {
-                name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
-                key  = "REDIS_HOST"
-              }
-            }
-          }
-
-          env {
-            name = "REDIS_PORT"
-            value_from {
-              secret_key_ref {
-                name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
-                key  = "REDIS_PORT"
-              }
-            }
-          }
-
-          env {
-            name = "JWT_SECRET"
-            value_from {
-              secret_key_ref {
-                name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
-                key  = "JWT_SECRET"
-              }
+          env_from {
+            secret_ref {
+              name = kubernetes_secret.csalgo_scheduler_env.metadata[0].name
             }
           }
 

--- a/csalgo-iac/modules/scheduler/variables.tf
+++ b/csalgo-iac/modules/scheduler/variables.tf
@@ -74,3 +74,15 @@ variable "jwt_secret" {
   type        = string
   sensitive   = true
 }
+
+variable "huggingface_api_token" {
+  description = "Huggingface API Token"
+  type        = string
+  sensitive   = true
+}
+
+variable "openai_api_token" {
+  description = "OpenAI API Token"
+  type        = string
+  sensitive   = true
+}

--- a/csalgo-iac/modules/server/main.tf
+++ b/csalgo-iac/modules/server/main.tf
@@ -17,6 +17,7 @@ resource "kubernetes_secret" "csalgo_server_env" {
     REDIS_PORT    = var.redis_port
     JWT_SECRET    = var.jwt_secret
     HUGGINGFACE_API_TOKEN = var.huggingface_api_token
+    OPENAI_API_TOKEN = var.openai_api_token
   }
 
   type = "Opaque"

--- a/csalgo-iac/modules/server/variables.tf
+++ b/csalgo-iac/modules/server/variables.tf
@@ -80,3 +80,9 @@ variable "huggingface_api_token" {
   type        = string
   sensitive   = true
 }
+
+variable "openai_api_token" {
+  description = "OpenAI API 토큰"
+  type        = string
+  sensitive   = true
+}

--- a/csalgo-iac/variables.tf
+++ b/csalgo-iac/variables.tf
@@ -81,6 +81,12 @@ variable "huggingface_api_token" {
   sensitive   = true
 }
 
+variable "openai_api_token" {
+  description = "OpenAI API Token"
+  type        = string
+  sensitive   = true
+}
+
 variable "server_image" {
   type = string
   default = "csalgo.kr.ncr.ntruss.com/csalgo-server:latest"

--- a/csalgo-scheduler/src/main/resources/application.yml
+++ b/csalgo-scheduler/src/main/resources/application.yml
@@ -69,3 +69,7 @@ external:
   resources:
     usage-url: ${USAGE_GUIDE_URL:https://kr.object.ncloudstorage.com/contest34-bucket-priv/resources/usage.html}
     images-url: ${IMAGE_BASE_URL:https://kr.object.ncloudstorage.com/contest34-bucket-priv/resources/images}
+
+openai:
+  token: ${OPENAI_API_TOKEN:dummy-token}
+  model: ${OPENAI_MODEL:gpt-4o-mini}

